<?xml version="1.0" ?>
<launch>
  <arg name="path" default="aep_office/stl" />

  <arg name="param_file" default="$(find rpl_exploration)/config/$(arg path).yaml" />


  <group ns="aeplanner">
    <node pkg="tf2_ros" type="static_transform_publisher" name="primesense_broadcaster" args="0.05 0 0.13 -1.57079633 0 -1.57079633 base_link camera_link" />

    <include file="$(find aeplanner)/launch/aeplanner_nodelet.launch"/>
    <include file="$(find pigain)/launch/pig_cpp.launch"/>
    <include file="$(find rpl_exploration)/launch/fly_to.launch"/>
    <include file="$(find rpl_exploration)/launch/octomap_server.launch"/>
    <include file="$(find rrtplanner)/launch/rrtplanner.launch">
      <arg name="path" value="$(arg path)" />
      <arg name="param_file" value="$(arg param_file)" />
    </include>


    <node name="rpl_exploration" pkg="rpl_exploration" type="rpl_exploration" output="screen" />
    <rosparam command="load" file="$(arg param_file)" />
  </group>

  <!-- Nodelet manager for camera-->
  <node pkg="nodelet" type="nodelet" args="manager"
name="camera_manager" output="screen"/>

  <node pkg="nodelet" type="nodelet" name="image_to_points" args="load depth_image_proc/point_cloud_xyz camera_manager --no-bond">
    <remap from="camera_info" to="/virtual_camera/depth/camera_info" />
    <remap from="image_rect" to="/virtual_camera/depth/image_raw" />
    <remap from="points" to="/virtual_camera/depth/points" />
    <param name="queue_size" value="100" />
  </node>

  <node pkg="rpl_exploration" type="image_maxing.py" name="image_maxing" output="screen" />

  <node pkg="stl_evaluation" type="stl_evaluation_node" name="std_evaluation_node" output="screen" required="true">
    <rosparam command="load" file="$(arg param_file)" />
    <!-- AEP apartment -->
    <!--
    <param name="bbx_min_x" value="-10.0" />
    <param name="bbx_min_y" value="-5.0" />
    <param name="bbx_min_z" value="0.3" />
    <param name="bbx_max_x" value="10" />
    <param name="bbx_max_y" value="5" />
    <param name="bbx_max_z" value="2.5" />
    <param name="volume_scaler" value="1.0" />
    -->
    <!-- AEP office -->
    <!-- 
    <param name="bbx_min_x" value="-27.0" />
    <param name="bbx_min_y" value="-3.0" />
    <param name="bbx_min_z" value="0.3" />
    <param name="bbx_max_x" value="3" />
    <param name="bbx_max_y" value="17" />
    <param name="bbx_max_z" value="2.5" />
    <param name="volume_scaler" value="1.0" />
     -->
     <!-- Garage -->
     <!--
    <param name="bbx_min_x" value="-10.0" />
    <param name="bbx_min_y" value="-3.0" />
    <param name="bbx_min_z" value="0.3" />
    <param name="bbx_max_x" value="24" />
    <param name="bbx_max_y" value="30.5" />
    <param name="bbx_max_z" value="2.5" />
    <param name="volume_scaler" value="1.0" />
    -->
    <!-- Office -->
    <!--
    <param name="bbx_min_x" value="-2.0" />
    <param name="bbx_min_y" value="-2.0" />
    <param name="bbx_min_z" value="0.3" />
    <param name="bbx_max_x" value="8" />
    <param name="bbx_max_y" value="10.0" />
    <param name="bbx_max_z" value="2.5" />
    <param name="volume_scaler" value="1.0" />
    -->
    <!-- Power plant -->
    <!--
    <param name="bbx_min_x" value="-1.0" />
    <param name="bbx_min_y" value="-24.0" />
    <param name="bbx_min_z" value="-4.0" />
    <param name="bbx_max_x" value="30" />
    <param name="bbx_max_y" value="7.0" />
    <param name="bbx_max_z" value="26" />
    <param name="volume_scaler" value="1.0" />
    -->
    <!-- Triangle small -->
    <!--
    <param name="bbx_min_x" value="-10.0" />
    <param name="bbx_min_y" value="-3.0" />
    <param name="bbx_min_z" value="0.3" />
    <param name="bbx_max_x" value="83.3" />
    <param name="bbx_max_y" value="90.3" />
    <param name="bbx_max_z" value="2.5" />
    <param name="volume_scaler" value="1.0" />
    -->
  </node>

  <!-- <node pkg="rosbag" type="record" name="rosbag" output="screen" args="-a -x /camera/(.*)|/virtual_camera/(.*) -o $(find stl_evaluation)/data/$(arg path)"/> -->
  <!-- <node pkg="rosbag" type="record" name="rosbag" output="screen" args="-o $(find stl_evaluation)/data/$(arg path) /aeplanner/free_cells_vis_array /aeplanner/ltl_path /aeplanner/occupied_cells_vis_array /aeplanner/pig_markers /aeplanner/rrt_path /aeplanner/rrtree /aeplanner/ltl_stats /stl_evaluation/volume /mavros/setpoint_position/local /mavros/local_position/pose"/> -->
  <!-- <node pkg="rosbag" type="record" name="rosbag" output="screen" args="-o $(find stl_evaluation)/bags/$(arg path) /aeplanner/ltl_path /aeplanner/occupied_cells_vis_array /aeplanner/pig_markers /aeplanner/rrt_path /aeplanner/rrtree /aeplanner/ltl_stats /stl_evaluation/volume /mavros/setpoint_position/local /mavros/local_position/pose /tf /tf_static"/> -->
</launch>
